client/state/initial-state.js

```diff
@@ -118,6 +118,13 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 	const throttledSaveState = throttle(
 		function() {
 			const nextState = reduxStore.getState();
+
+			// Remove properties that shouldn't be persisted for logged-out
+			// users.
+			if ( ! isLoggedIn() ) {
+				delete nextState.currentUser;
```

Comment:
Could we try moving this to `loadInitialState` when we're loading state from `redux-state-logged-out`? I think we can do this one time, unless some user id is being set incorrectly in redux for logged out folks.

Reply:
Let's also add a test case in `state/test/initial-state.js`

Reply:
also how about not using `delete` but instead creating a new copy of the state? notice how _the very next lines_ should return from this function without updating anything because we thought that the two were equal

```js
const prevState = reduxStore.getState();
const { currentUser, ...loggedOutState } = prevState;

const nextState = isLoggedIn()
	? prevState
	: loggedOutState;

if ( state && nextState === state ) {
	return;
}
```

Reply:
> also how about not using delete but instead creating a new copy of the state?

Good point - we probably shouldn't be modifying the main state object like that.

(However, now that we're looking to move to a different implementation, this issue probably won't come up again there.)